src/network/networkd-netlabel.c

```diff
@@ -0,0 +1,194 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include "netlink-util.h"
+#include "networkd-address.h"
+#include "networkd-link.h"
+#include "networkd-manager.h"
+#include "networkd-netlabel.h"
+#include "networkd-network.h"
+
+static int netlabel_handler(sd_netlink *rtnl, sd_netlink_message *m, Link *link) {
+        int r;
+
+        assert_se(rtnl);
+        assert_se(m);
+        assert_se(link);
+
+        r = sd_netlink_message_get_errno(m);
+        if (r < 0) {
+                log_link_message_warning_errno(link, m, r, "NetLabel operation failed, ignoring");
+                return 1;
+        }
+
+        log_link_debug(link, "NetLabel operation successful");
```

Comment:
Hmm, this message is not very informative. If I saw it without context, I would have no idea what was being done and why. Can you rephrase it?

Reply:
I agree it would be nice to have more verbose message explaining what exactly was successful (or what failed). But there isn't any context related to the operation available in the callback, just `Link` (which may have several `Address`) and the raw message.

In the earlier version I used synchronous sending and then the errors messages were able to be more precise.